providers/core.tcpsocket.js

```diff
@@ -75,15 +75,10 @@ Socket_chrome.prototype.connect = function(hostname, port, cb) {
   chrome.sockets.tcp.create({}, function(createInfo) {
     this.id = createInfo.socketId;
     try {
-      chrome.sockets.tcp.connect(this.id, hostname, port, function (result) {
-        if (result < 0) {
-          var errorString = Socket_chrome.errorStringOfCode(result);
-          var freedomErrorCode = Socket_chrome.freedomErrorCode(errorString);
-          cb(undefined, {
-            'errcode': freedomErrorCode,
-            'message': 'Chrome Connection Failed: ' + errorString +
-                ' ' + chrome.runtime.lastError.message
-          });
+      chrome.sockets.tcp.connect(this.id, hostname, port, function(result) {
+        var error = Socket_chrome.chromeErrorHandler(result);
```

Comment:
Also, the change in this particular spot isn't an actual logic change, just organizational. The error is calculated from the result code with the exact same logic as before, just in the `Socket_chrome.chromeErrorHandler` method. The only reason I even edited this spot was to avoid repeating code, i.e. pull out that logic and invoke it from https://github.com/freedomjs/freedom-for-chrome/pull/75/files#diff-c02bde11d55bcda45d0a199ac895aef8R513 which is where I'm actually trying to fix the issue.

`chromeErrorHandler` does potentially use `chrome.runtime.lastError` (https://github.com/freedomjs/freedom-for-chrome/pull/75/files#diff-c02bde11d55bcda45d0a199ac895aef8R341), though granted it only examines it if the result code says there is a problem (i.e. doesn't check for its existence as an indicator of an error). That said, this spot may still be wrong - let me know if you think that `chrome.runtime.lastError` still needs to be separately examined.

And overall, where in uProxy did you encounter this error? I'm happy to take a look at it in context as well. Thanks!